lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,355 @@
+from flask_restful import Resource, abort
+from flask import request, make_response, jsonify
+from pbench.server.database.models.metadata import Metadata, MetadataKeys
+from pbench.server.api.auth import Auth
+
+
+class UserMetadata(Resource):
+    """
+    Abstracted pbench API for handling user metadata by using graphql schema
+    """
+
+    def __init__(self, config, logger, auth):
+        self.server_config = config
+        self.logger = logger
+        self.auth = auth
+
+    @Auth.token_auth.login_required(optional=True)
+    def post(self, key):
+        """
+        Post request for creating metadata instance for a user.
+
+        The url requires a metadata session key such as /metadata/<key>
+        the only keys acceptable to create the metadata sessions are favourite/saved
+
+        This requires a JSON data with required user metadata fields
+        {
+            "value": "blog text" <Client defined text, can be a string of json>,
+        }
+        Example: {"value": '{"config": "config", "description": "description"}'}
+
+        Authorization header can be included as
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+        If the authorization header is not present, the created metadata session becomes public by default
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success"
```

Comment:
I really don't like this "message"/"data" response format. The "success" is redundant with the HTTP status code, and a complete waste of bytes. (Notwithstanding Aquib's discoveries about the quirks of umi wrappers.) I'd rather just return the object. Basically, HTTP has standard protocol for this sort of information, and I don't see how it makes sense to duplicate that in the response payload. A more detailed explanation on a failure is a completely different matter; but on success the only required "detailed explanation" is the object itself.

Reply:
yes, fixed